presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/HivePrincipal.java

```diff
@@ -13,61 +13,66 @@
  */
 package io.prestosql.plugin.hive.metastore;
 
-import org.apache.hadoop.hive.metastore.api.PrincipalType;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.prestosql.spi.security.PrestoPrincipal;
+import io.prestosql.spi.security.PrincipalType;
 
-import javax.annotation.concurrent.Immutable;
-
-import java.util.Locale;
 import java.util.Objects;
+import java.util.Set;
 
-import static com.google.common.base.MoreObjects.toStringHelper;
-import static org.apache.hadoop.hive.metastore.api.PrincipalType.ROLE;
-import static org.apache.hadoop.hive.metastore.api.PrincipalType.USER;
+import static com.google.common.collect.ImmutableSet.toImmutableSet;
+import static java.util.Locale.ENGLISH;
+import static java.util.Objects.requireNonNull;
 
-@Immutable
+/**
+ * In Hive role names are case insensitive and user names are case sensitive.
```

Comment:
This is not a javadoc for this class.

Reply:
Bump. This comment does not explain what this class represents or is doing, hence should not be the first sentence of the javadoc. 

i suggest that you remove this comment